lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
> `$tree = {$lang => $tree}`

Does YAML expect language to be in a certain format? My understanding is that it expects locale, not raw language (e.g. `pt_br` instead of `pt-br`). I think we need to explicitly format `$lang` here. [Serge::Util::locale_from_lang](https://github.com/evernote/serge/blob/master/lib/Serge/Util.pm#L162) might help. Probably you will need further lowercase it.